gatsby-node.js

```diff
@@ -42,6 +42,107 @@ const getMessages = (path, language) => {
+ * @param {string} path filepath for translated mdx file
+ * @returns boolean for if file is outdated or not
+ */
+const checkIsMdxOutdated = (path) => {
+  try {
+    const splitPath = path.split("/")
+    splitPath.splice(7, 2)
+    const englishPath = splitPath.join("/")
+
+    const re = /([#]+) [^\{]+\{#([^\}]+)\}/gim
+
+    const translatedData = fs.readFileSync(path, "utf-8")
+    const englishData = fs.readFileSync(englishPath, "utf-8")
+
+    let englishMatch = ""
+    let intlMatch = ""
+    englishData.match(re).forEach((match) => {
+      englishMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+    translatedData.match(re).forEach((match) => {
+      intlMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+
+    return englishMatch !== intlMatch
+  } catch (err) {
+    return true
+  }
+}
+
+/**
+ * JSON isOutdated check
+ * Checks if translated JSON file exists.
+ * If translated file exists, checks that all translations are present (checks keys), and that all the keys are the same.
+ * If translation file exists, isContentEnglish will be false
+ * @param {*} path url path used to derive file path from
+ * @param {*} lang language abbreviation for language path
+ * @returns {{isOutdated: boolean, isContentEnglish: boolean}}
+ */
+const checkIsPageOutdated = async (path, lang) => {
+  try {
+    // Files that need index appended on the end. Ex page-index.json, page-developers-index.json, page-eth2-index.json
+    const indexFilePaths = ["", "developers", "eth2"]
+    const filePath = path.split("/").filter((text) => text !== "")
+
+    if (
+      indexFilePaths.includes(filePath[filePath.length - 1]) ||
+      filePath.length === 0
+    ) {
+      filePath.push("index")
+    }
+
+    const joinedFilepath = filePath.join("-")
+    const srcPath = `${__dirname}/src/intl/${lang}/page-${joinedFilepath}.json`
+    const englishPath = `${__dirname}/src/intl/en/page-${joinedFilepath}.json`
+
+    // If no file exists, default to english
+    if (!fs.existsSync(srcPath)) {
+      return {
+        isOutdated: true,
+        isContentEnglish: true,
+      }
+    } else {
+      const translatedData = JSON.parse(fs.readFileSync(srcPath))
+      const englishData = JSON.parse(fs.readFileSync(englishPath))
+      const translatedKeys = Object.keys(translatedData)
+      const englishKeys = Object.keys(englishData)
+
+      // Check if same amount of keys
+      if (translatedKeys.length !== englishKeys.length) {
+        return {
+          isOutdated: true,
+          isContentEnglish: false,
+        }
+      }
+
+      // Check if all the keys are the same
+      if (
+        JSON.stringify(translatedKeys.sort()) !==
+        JSON.stringify(englishKeys.sort())
+      ) {
+        return {
+          isOutdated: true,
+          isContentEnglish: false,
+        }
+      }
+
+      return {
+        isOutdated: false,
+        isContentEnglish: false,
+      }
+    }
+  } catch {
+    return true
```

Comment:
I don't like this return that is absorbing all the errors and is different from the return type from the function.

Let the function throw its errors and that the consumer handles what to do with it.